lib/octokit/repository.rb

```diff
@@ -10,7 +10,11 @@ class Repository
     #
     # @return [Repository]
     def self.from_url(url)
-      Repository.new(URI.parse(url).path[1..-1])
+      repo = URI.parse(url).path[1..-1]
+      if /repos\/(?<owner>[\w.-]+)\/(?<name>[\w.-]+)/i =~ repo
```

Comment:
Can we move this to the top of the class as a constant? `REPOSITORY_API_URL_PATTERN` maybe?

From looking at #704 I think @aroben was hoping to match on a web site URL, too (`/octokit/octokit.rb/pull/1` vs `/repos/octokit/octokit.rb/pulls/1`). That gets tricky because we can't trust every URL to be a repository, but maybe we don't care.